tas-client/src/tas-client/FeatureProvider/TasApiFeatureProvider.ts

```diff
@@ -38,16 +39,48 @@ export class TasApiFeatureProvider extends FilteredFeatureProvider {
         }
 
         //axios webservice call.
-        let response: AxiosResponse<TASFeatureData> = await this.httpClient.get({ headers: headers });
+        let response: AxiosResponse<TASFeatureData> | undefined;
+
+        try {
+            /**
+             * As mentioned in the axios docs:
+             * Axios Promise will handle handle the catch
+             * section of promises whenever a request is not succesful.
+             * https://axios-http.com/docs/handling_errors
+             */
+            response = await this.httpClient.get({ headers: headers });
+        } catch (error) {
+            const axiosError = error as AxiosError;
+            const properties: Map<string, string> = new Map();
+            if (axiosError.response) {
+                // The request was made and the server responded with a status code
+                // that falls out of the range of 2xx
+                properties.set('ErrorType', 'ServerError');
+            } else if (axiosError.request) {
+                // The request was made but no response was received
+                // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+                // http.ClientRequest in node.js
+                properties.set('ErrorType', 'NoResponse');
+            } else {
+                // Something happened in setting up the request that triggered an Error
+                properties.set('ErrorType', 'GenericError');
+            }
+            this.telemetry.postEvent(TASAPI_FETCHERROR_EVENTNAME, properties);
+        }
+
+        // In case the response fetching failed, throw
+        // exception so that the caller exits.
+        if (!response) {
+            throw Error(TASAPI_FETCHERROR_EVENTNAME);
+        }
 
         // If we have at least one filter, we post it to telemetry event.
         if (filters.keys.length > 0) {
             this.PostEventToTelemetry(headers);
```

Comment:
@daytonellwanger this is the method I'm referring to: PostEventToTelemetry, do you think we should be using this wrapper instead? Or do we want to just send the Error property in the telemetry event?

Reply:
I'm leaning towards a new event and not re-using that wrapper. It seems like a sufficiently different use case with different requirements.

Reply:
@daytonellwanger sounds good, I'll leave it as it is then.
Let me know if there's anything else that needs to be addressed.